src/Transports.AspNetCore/IUserContextBuilder.cs

```diff
@@ -1,16 +1,17 @@
+#nullable enable
+
 using Microsoft.AspNetCore.Http;
 
 namespace GraphQL.Server.Transports.AspNetCore;
 
 /// <summary>
-/// Interface which is responsible of building a UserContext for a GraphQL request
+/// Creates a user context from a <see cref="HttpContext"/>.
+/// <br/><br/>
+/// The generated user context may be used for one or more GraphQL requests or
```

Comment:
>  for one or more GraphQL requests

in case of batched requests?

Reply:
Yes.

One thing we should think about is altering the interface for building the user context to also contain the payload from the WebSocket initialization message. This would require a redesign of a little of my WebSocket code. One additional benefit is that it could occur after authorization, which it doesn’t now. I’ll probably make that change in my code shortly. We could also add a property to the options to specify a single user context instance per request probably (need to check the subscription implications).

Reply:
I can not say anything about this. I do not know WebSockets' stuff.